.jenkins/pipelines/pipeline-tests.Jenkinsfile

```diff
@@ -0,0 +1,185 @@
+/* A Jenkins pipeline that will handle code coverage and nightly tests
+*  These are the original pipelines:
+*  https://github.com/deislabs/mystikos/blob/main/.azure_pipelines/ci-pipeline-code-coverage-nightly.yml
+*/
+
+pipeline {
+    agent {
+        label 'ACC-1804-DC8-build-machine'
+    }
+    options {
+        timeout(time: 300, unit: 'MINUTES')
+    }
+    parameters {
+        string(name: "REPOSITORY", defaultValue: "deislabs")
+        string(name: "BRANCH", defaultValue: "main", description: "Branch to build")
+        choice(name: "TEST_CONFIG", choices:['Nightly', 'Code Coverage'], description: "Test configuration to execute")
+        choice(name: "REGION", choices:['useast', 'canadacentral'], description: "Azure region for SQL test")
+    }
+    environment {
+        MYST_SCRIPTS =      "${WORKSPACE}/scripts"
+        JENKINS_SCRIPTS =   "${WORKSPACE}/.jenkins/scripts"
+        MYST_NIGHTLY_TEST = 1
+        MYST_ENABLE_GCOV =  1
+        BUILD_RESOURCES =   "build-resources-${GIT_COMMIT[0..7]}"
+        BUILD_USER = sh(
+            returnStdout: true,
+            script: 'echo \${USER}'
+        )
+    }
+    stages {
+        stage("Initialize Workspace") {
+            steps {
+                sh """
+                   ${JENKINS_SCRIPTS}/global/clean-temp.sh
+                   """
+                azureDownload(
```

Comment:
This is the first stage of a new PR build right? What build resources is there (or should be) downloaded?

Reply:
Is this so one can rerun jobs and skip the build phase?

Reply:
Yeah, exactly!